python/lsst/resources/http.py

```diff
@@ -601,6 +859,42 @@ def _openImpl(
             with super()._openImpl(mode, encoding=encoding) as http_handle:
                 yield http_handle
 
+    def _dump_response(self, resp: requests.Response) -> None:
+        """Logs the contents of a HTTP or webDAV request and its response.
+
+        Parameters
+        ----------
+        resp : `requests.Response`
+            The response to log.
+
+        Notes
+        -----
+        Intended for development purposes only. To use do:
+
+            self._dump_response(resp)
+        """
+        log.debug("-----------------------------------------------")
+        log.debug("Request")
+        log.debug(f"   method={resp.request.method}")
+        log.debug(f"   URL={resp.request.url}")
+        log.debug(f"   headers={resp.request.headers}")
+        if resp.request.method == "PUT":
+            log.debug("   body=<data>")
+        elif resp.request.body is None:
+            log.debug("   body=<empty>")
+        else:
+            log.debug(f"   body={resp.request.body[:120]!r}")
+
+        log.debug("Response:")
+        log.debug(f"   status_code={resp.status_code}")
+        log.debug(f"   headers={resp.headers}")
+        if resp.content is None:
```

Comment:
mypy is convinced that `resp.content` can never be `None`. I don't know if that's true but maybe changing this line to `if not resp.content` would work.

Reply:
I think mypy is wrong, indeed. I changed the test to make it happy.

Reply:
The problem with requests type annotations is that they come in through the `types-requests` package which is a standalone external package not directly part of the `requests` code base. It's possible there is a bug in that annotation.

Reply:
My issue is that the results of running mypy locally in my development host are not the same as the ones ran in github actions. That makes I need several roundtrips to make mypy happy. I need to take the time to understand in detail how to make it work locally.

Reply:
are you using the mypy from  `rubin-env-developer` conda package? I seem to have 0.991 installed and that is the version the action seems to be using.